File: 01_Array_Hashings/09_Longest_Consecutive_Sequence/128_Longest_Consecutive_Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)

        streak_max = 0
        for n in nums:
            if n - 1 not in nums_set:
                print(f" {n} is the first ele of sequence")
                streak = 1
                while n+1 in nums_set:
                    n += 1
                    streak += 1
                if streak > streak_max: 
                    streak_max = streak
        
        return streak_max  

File: 01_Array_Hashings/09_Longest_Consecutive_Sequence/128_Longest_Consecutive_Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> nums_set(nums.begin(), nums.end());

        int max_streak = 0;
        for (int n: nums){
            bool is_in = nums_set.find(n-1) != nums_set.end();
            if (nums_set.find(n-1) == nums_set.end()){
                cout << n << "is first ele";
                int streak = 1;
                while (nums_set.find(n+1) != nums_set.end()){
                    streak++;
                    n++;
                }

                streak > max_streak ? max_streak = streak : max_streak;
            }
        }
    return max_streak;

    }
};
